athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchQueryUtils.java

```diff
@@ -0,0 +1,242 @@
+/*-
+ * #%L
+ * athena-elasticsearch
+ * %%
+ * Copyright (C) 2019 - 2020 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.elasticsearch;
+
+import com.amazonaws.athena.connector.lambda.domain.predicate.EquatableValueSet;
+import com.amazonaws.athena.connector.lambda.domain.predicate.Range;
+import com.amazonaws.athena.connector.lambda.domain.predicate.ValueSet;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.fetch.subphase.FetchSourceContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class has interfaces used for the generation of projections and predicates used for document search queries.
+ */
+class ElasticsearchQueryUtils
+{
+    private static final Logger logger = LoggerFactory.getLogger(ElasticsearchQueryUtils.class);
+
+    // Predicate conjunctions.
+    private static final String AND_OPER = " AND ";
+    private static final String OR_OPER = " OR ";
+    private static final String NO_OPER = "";
+    private static final String NOT_OPER = "NOT ";
```

Comment:
will not having a leading space cause issues when combined with other operators?

Reply:
Should be okay. NOT is only used for one case (exclusive equitable values e.g. "NOT field:(value1 OR value2...)" If multiple predicates are combined they will most likely be combined with the AND_OPER which has a trailing blank (e.g. "AND NOT...") OR_OPER has a trailing blank as well.